packages/@aws-cdk/aws-batch/lib/compute-environment.ts

```diff
@@ -347,37 +361,42 @@ export class ComputeEnvironment extends Resource implements IComputeEnvironment
 
     // Only allow compute resources to be set when using MANAGED type
     if (props.computeResources && this.isManaged(props)) {
+      const isFargate = ComputeResourceType.FARGATE === props.computeResources.type
+        || ComputeResourceType.FARGATE_SPOT === props.computeResources.type;
+
```

Comment:
This exact code is repeated in `validateProps` - lets just compute this once and have `validateProps` accept an `isFargate` boolean as well.

Reply:
Fixed

Reply:
Done

Reply:
Fixed